benchmark/clblast/utils.hpp

```diff
@@ -13,6 +13,35 @@ void create_benchmark(blas_benchmark::Args &args, ExecutorType *exPtr);
 
 namespace utils {
 
+inline void print_device_information(cl_device_id device) {
+  char device_name[64];
+  cl_device_type *device_type = new cl_device_type;
+  clGetDeviceInfo(device, CL_DEVICE_NAME, 64 * sizeof(char), device_name, NULL);
```

Comment:
- make this `nullptr`, please
- There is some code duplication here: The size of the array you are writing to is stated here and at the point of declaration above. It's better to use  `sizeof(device_name)`. It would be even better to use `std::string` here because the length of the string might exceed 64 bytes:

```c++
size_t device_name_length = 0;
clGetDeviceInfo(device, CL_DEVICE_NAME, 0, nullptr, &device_name_length);
if(!device_name_length) {
    return;
}
std::string device_name(device_name_length);
clGetDeviceInfo(device, CL_DEVICE_NAME, 0, std::addressof(device_name.front()), nullptr);
```

Reply:
Thanks. Your code doesn't work but I've used a slight variation of it.